File: PS-Sprint-1/Main_1to50/_04_Armstrong_Numbers/_04_Armstrong_Numbers.js
let num = 153;
let arm = 0,
  copy,
  rem;
copy = num;

while (num > 0) {
  rem = num % 10;
  console.log("Remainder:", rem);
  
  arm += (rem * rem * rem);
  console.log("Current Armstrong sum:", arm);
  
  num = Math.floor(num / 10); // Use Math.floor to get integer part
  console.log("Updated number:", num);
}

if (copy === arm) {
  console.log(copy + " is an Armstrong Number");
} else {
  console.log(copy + " is not an Armstrong Number");
}

File: PS-Sprint-1/Main_1to50/_04_Armstrong_Numbers/_04_Armstrong_Numbers.java
import java.util.Scanner;

public class _04_Armstrong_Numbers {
     public static void main(String[] args) {

          Scanner sc = new Scanner(System.in);
          int rem, arm = 0;
          int c;

          System.out.println("Enter your Number");
          int numb = sc.nextInt();
          c = numb;
          while (numb > 0) {

               rem = numb % 10;
               arm = (rem * rem * rem)+arm;
               numb = numb / 10;

          }
          if (c==arm) {
               System.out.println("Armstrong numbers");
          } else {
               System.out.println(" not Armstrong number");
          }

     }

}
